src/PTFAnalysis.cpp

```diff
@@ -584,35 +584,37 @@ PTFAnalysis::PTFAnalysis( TFile* outfile, Wrapper & wrapper, double errorbar, PT
   bool pulse_location_cut;
   bool fft_cut;
   bool do_pulse_finding;
-  bool dofit = true;
+  bool dofit;
 
   config.Load(config_file);
-  if( !config.Get("terminal_output", terminal_output) ){
-    cout << "Missing terminal_output parameter from config file." << endl;
-    exit( EXIT_FAILURE );
-  }
-  if( !config.Get("pulse_location_cut", pulse_location_cut) ){
-    cout << "Missing pulse_location_cut parameter from config file." << endl;
-    exit( EXIT_FAILURE );
-  }
-  if( !config.Get("fft_cut", fft_cut) ){
-    cout << "Missing fft_cut parameter from config file." << endl;
-    exit( EXIT_FAILURE );
-  }
-  if( !config.Get("do_pulse_finding", do_pulse_finding) ){
-    cout << "Disabling pulse finding." << std::endl;
-    do_pulse_finding = false;
-  }
-  if( !config.Get("do_pulse_fitting", dofit) ){
-    std::cout <<"Disabling pulse fitting"<< std::endl;
-    dofit = true;
-  }else{
-    if(dofit){
-      std::cout <<"Enabling pulse fitting"<< std::endl;
-    }else{
-      std::cout <<"Disabling pulse fitting"<< std::endl;
+    if( !config.Get("terminal_output", terminal_output) ){
+      cout << "Missing terminal_output parameter from config file." << endl;
+      exit( EXIT_FAILURE );
+    }
+    if( !config.Get("pulse_location_cut", pulse_location_cut) ){
+      cout << "Missing pulse_location_cut parameter from config file." << endl;
+      exit( EXIT_FAILURE );
     }
+    if( !config.Get("fft_cut", fft_cut) ){
+      cout << "Missing fft_cut parameter from config file." << endl;
+      exit( EXIT_FAILURE );
   }
+    if( !config.Get("do_pulse_finding", do_pulse_finding) ){
+      cout << "Disabling  pulse finding." <<do_pulse_finding << std::endl;
+      do_pulse_finding = false;
+     }
+    //cout<<do_pulse_finding<<endl;                                                                                                                                                                                                             
+    if( !config.Get("do_pulse_fitting", dofit) ){
+      dofit = false;
+      std::cout <<"Disabling pulse fitting"<<dofit<< std::endl;
+
+    }
+    else{
+    //  if(dofit){                                                                                                                                                                                                                              
+    dofit = true;
```

Comment:
This seems wrong.  The line 

config.Get("do_pulse_fitting", dofit)

has already set dofit to what is in the config.dat file.  But this 'dofit=true' means that you are going to do the pulse fitting even if 

do_pulse_fitting = false

in the config.dat file.  You should let do_pulse_fitting be set from the config.dat file